doc/internal/man3/evp_generic_fetch.pod

```diff
@@ -151,9 +156,16 @@ And here's the implementation of the FOO method fetcher:
 
     EVP_FOO_meth_free(EVP_FOO *foo)
     {
+        int i;
+
         if (foo != NULL) {
             OSSL_PROVIDER *prov = foo->prov;
 
+            CRYPTO_DOWN_REF(&foo->refcnt, &i);
+            if (i > 0)
+                return;
+
+            CRYPTO_FREE_REF(&foo->refcnt);
             OPENSSL_free(foo);
             ossl_provider_free(prov);
```

Comment:
Does the above code work with the CRYPTO_DOWN_REF in ossl_provider_free

Reply:
I think so, this was the basis of your earlier comment.

Reply:
Both the provider and the EVP_FOO have a reference count, each free method decreasing it's own count. So I don't see what could be wrong.

I do wonder why ossl_provder_free() is not called earlier, before OPENSSL_free() is called. It would avoid taking a copy of the foo->prov pointer.